addons/purchase_edi_ubl_bis3/models/purchase_edi_common.py

```diff
@@ -0,0 +1,184 @@
+    'SE': {'0007': 'vat'},
+    'SI': {'9949': 'vat'},
+    'SK': {'9950': 'vat'},
+    'SM': {'9951': 'vat'},
+    'TR': {'9952': 'vat'},
+    'VA': {'9953': 'vat'},
+}
+
+
+class PurchaseEdiCommon(models.AbstractModel):
+    _name = "purchase.edi.common"
+    _description = "Common functions for EDI documents"
+
+    # -------------------------------------------------------------------------
+    # HELPERS
+    # -------------------------------------------------------------------------
+
+    def format_float(self, amount, precision_digits):
+        if amount is None:
+            return None
+        return float_repr(float_round(amount, precision_digits), precision_digits)
+
+    def _get_currency_decimal_places(self, currency_id):
+        # Allows other documents to easily override in case there is a flat max precision number
+        return currency_id.decimal_places
+
+    def _get_uom_unece_code(self, line):
+        """
+        list of codes: https://docs.peppol.eu/poacc/billing/3.0/codelist/UNECERec20/
+        or https://unece.org/fileadmin/DAM/cefact/recommendations/bkup_htm/add2c.htm (sorted by letter)
+        """
+        xmlid = line.product_uom.get_external_id()
+        if xmlid and line.product_uom.id in xmlid:
+            return UOM_TO_UNECE_CODE.get(xmlid[line.product_uom.id], 'C62')
+        return 'C62'
+
+    def _find_value(self, xpath, tree):
+        # avoid 'TypeError: empty namespace prefix is not supported in XPath'
+        nsmap = {k: v for k, v in tree.nsmap.items() if k is not None}
+        return find_xml_value(xpath, tree, nsmap)
+
+    # -------------------------------------------------------------------------
+    # TAXES
+    # -------------------------------------------------------------------------
+
+    def _get_tax_unece_codes(self, order, tax):
+        """
+        Source: doc of Peppol (but the CEF norm is also used by factur-x, yet not detailed)
+        https://docs.peppol.eu/poacc/billing/3.0/syntax/ubl-invoice/cac-TaxTotal/cac-TaxSubtotal/cac-TaxCategory/cbc-TaxExemptionReasonCode/
+        https://docs.peppol.eu/poacc/billing/3.0/codelist/vatex/
+        https://docs.peppol.eu/poacc/billing/3.0/codelist/UNCL5305/
+        :returns: {
+            tax_category_code: str,
+            tax_exemption_reason_code: str,
+            tax_exemption_reason: str,
+        }
+        """
+        def create_dict(tax_category_code=None, tax_exemption_reason_code=None, tax_exemption_reason=None):
+            return {
+                'tax_category_code': tax_category_code,
+                'tax_exemption_reason_code': tax_exemption_reason_code,
+                'tax_exemption_reason': tax_exemption_reason,
+            }
+
+        supplier = order.partner_id
+        customer = order.company_id.partner_id.commercial_partner_id
+
+        # add Norway, Iceland, Liechtenstein
+        european_economic_area = self.env.ref('base.europe').country_ids.mapped('code') + ['NO', 'IS', 'LI']
```

Comment:
Okay maybe it's a bit too much and very Europe specific. We could create a hook method for future localization and put a simple value?